File: csharp/MapleStory.OpenAPI/Src/Dto/Union/UnionChampionInfoDTO.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;

namespace MapleStory.OpenAPI.Dto
{
    /// <summary>
    /// 유니온 챔피언 상세 정보

    /// </summary>
    public class UnionChampionInfoDTO
    {
        /// <summary>
        /// 유니온 챔피언 슬롯
        /// </summary>
        [JsonProperty("champion_slot")]
        public int ChampionSlot { get; set; }

        /// <summary>
        /// 유니온 챔피언 등급
        /// </summary>
        [JsonProperty("champion_grade")]
        public string ChampionGrade { get; set; }

        /// <summary>
        /// 유니온 챔피언 캐릭터의 직업
        /// </summary>
        [JsonProperty("champion_class")]
        public string ChampionClass { get; set; }

        /// <summary>
        /// 챔피언 휘장 효과
        /// </summary>
        [JsonProperty("champion_badge_info")]
        public List<UnionChampionBadgeInfoDTO> ChampionBadgeInfo { get; set; }
    }
}

File: java/src/main/java/dev/spiralmoon/maplestory/api/msea/dto/union/UnionChampionInfoDTO.java
package dev.spiralmoon.maplestory.api.msea.dto.union;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.List;

/**
 * Union Champion information
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class UnionChampionInfoDTO implements dev.spiralmoon.maplestory.api.common.dto.union.UnionChampionInfoDTO<UnionChampionBadgeInfoDTO> {

    /**
     * Union Champion character name
     */
    @JsonProperty("champion_name")
    private String championName;

    /**
     * Union Champion slot
     */
    @JsonProperty("champion_slot")
    private int championSlot;

    /**
     * Union Champion level
     */
    @JsonProperty("champion_grade")
    private String championGrade;

    /**
     * Union Champion character's job
     */
    @JsonProperty("champion_class")
    private String championClass;

    /**
     * Union Champion Insignia information
     */
    @JsonProperty("champion_badge_info")
    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<UnionChampionBadgeInfoDTO> championBadgeInfo;
}
